src/main/webapp/js/query.js

```diff
@@ -253,3 +257,49 @@ function loadImage(photoUrl) {
     img.src = photoUrl;
     photoEl.appendChild(img);
 }
+
+/*=========================
+    Directions to the Restaurant
+ =========================*/
+ let directionsRenderer;
+ let directionsService;
+function addMapScript() {
+    let script = document.createElement('script');
+    script.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyDbEPugXWcqo1q6b-X_pd09a0Zaj3trDOw&callback=initMap';
+    script.defer = true;
+    script.async = true;
+    document.head.appendChild(script);
+}
+
+function initMap() {
+  directionsRenderer = new google.maps.DirectionsRenderer();
+  directionsService = new google.maps.DirectionsService();
+  let lat = localStorage.getItem("lat")
+  let lng = localStorage.getItem("lng")
+  var map = new google.maps.Map(document.getElementById("map"), {
+    zoom: 16,
+    center: { lat: parseFloat(lat), lng: parseFloat(lng)}
+  });
+  directionsRenderer.setMap(map);
+  directionsRenderer.setPanel(document.getElementById("directionsPanel"));
+  calculateAndDisplayRoute(directionsService, directionsRenderer);
+}
+
+function calculateAndDisplayRoute(directionsService, directionsRenderer) {
+  let start = localStorage.getItem("lat") + "," + localStorage.getItem("lng");
+  let end = localStorage.getItem("restaurantAddress");
+  directionsService.route(
+    {
+      origin: start,
+      destination: end,
+      travelMode: "DRIVING"
+    },
+    (response, status) => {
+      if (status === "OK") {
+        directionsRenderer.setDirections(response);
+      } else {
+        window.alert("Directions request failed due to " + status);
```

Comment:
the status should be changed to a human readable one - as in understandable by the end user. Also window.alert is rather intrusive

Reply:
logged the statement